quinn-proto/src/connection/mod.rs

```diff
@@ -1371,13 +1393,21 @@ impl Connection {
             }
         }
 
+        let mut timestamp_iter = self.config.ack_timestamp_config.as_ref().map(|cfg| {
+            let decoder = ack.timestamp_iter(cfg.basis, cfg.exponent.0).unwrap();
+            let mut v: tinyvec::TinyVec<[PacketTimestamp; 10]> = tinyvec::TinyVec::new();
```

Comment:
Nit: import `TinyVec` directly, and avoid type annotations. The code here is looking pretty unidiomatic... Can use `rev()` to reverse the direction on the source iterator directly which should be faster, and `collect()` into the `TinyVec` (are you sure we need to allocate here instead of yielding the iterator directly)?

Reply:
It doesn't seem possible to `rev()` on the source iterator because its yielding elements directly from the frame. It's similar to the `AckIter` implementation where each iteration yields a value as it reads from the `Ack` frame. The values yielded by the `AckIter` are added to an `ArrayRangeSet` which is space optimized for handling ranges, and is backed by a `TinyVec`.

What I'm essentially trying to do is match a packet number and it's timestamp (if available) together by "zipping" both together. When reading the values from the frame (for both packet number and timestamp), the values are produced from high packet number to low, but when we use them later on in the code, they're used from low to high, hence why I do a reverse.

One option to reduce an allocation and avoid the need of `reverse`ing is to attempt to zip the packet number and timestamps together as the values are yielded when reading from the frame. However, the issue that I run into is that the `array_range_set` is space optimized so it only records the ranges. If I wanted to zip the packet number and timestamps together, the `array_range_set` would have to be modified to associate a packet number value to a timestamp; example tuple `(packet_number, Option<timestamp>)`.

However, I don't think it makes sense to modify the existing `array_range_set` for this draft, so I opted to handle it this way, which is to read all the timestamps value into a Vec, and then handle the timestamp and packet_number matching later in the code.

Reply:
Okay nvm, I think I found a better solution - going to test it and report back..